src/lib/cli.js

```diff
@@ -31,16 +31,34 @@ process.title = 'verdaccio';
 const logger = require('./logger');
 logger.setup(); // default setup
 
+const envinfo = require('envinfo');
 const commander = require('commander');
 const pkgVersion = module.exports.version;
 const pkgName = module.exports.name;
 
 commander
+  .option('--info', 'prints debugging information about the local environment')
   .option('-l, --listen <[host:]port>', 'host:port number to listen on (default: localhost:4873)')
   .option('-c, --config <config.yaml>', 'use this configuration file (default: ./config.yaml)')
   .version(pkgVersion)
   .parse(process.argv);
 
+if (commander.info) {
+	// eslint-disable-next-line no-console
+	console.log('\nEnvironment Info:');
+	envinfo
+	.run({
+	    System: ['OS', 'CPU'],
+	    Binaries: ['Node', 'Yarn', 'npm', 'docker'],
+	    Browsers: ['Chrome', 'Edge', 'Firefox', 'Safari'],
+	    npmGlobalPackages: ['verdaccio'],
+	})
+	.then(data => {
```

Comment:
There is a problem with this implementation, it is asynchronous, if you run it like this you will get this

```
verdaccio.master.git on   (git)-[pr/1365|cherry-or-revert]- [$!] is 📦 v4.0.4-fix-info via ⬢ v10.15.0 on 🐳 v19.03.0-rc2 at ☸️  docker-desktop took 3s 
➜ verdaccio --info

Environment Info:
 warn --- config file  - /Users/user/.config/verdaccio/config.yaml
 warn --- Plugin successfully loaded: verdaccio-htpasswd
 warn --- Plugin successfully loaded: verdaccio-audit
 warn --- http address - http://localhost:4873/ - verdaccio/4.0.4-fix-info

  System:
    OS: macOS 10.14
    CPU: (4) x64 Intel(R) Core(TM) i5-6267U CPU @ 2.90GHz
  Binaries:
    Node: 10.15.0 - ~/.nvm/versions/node/v10.15.0/bin/node
    Yarn: 1.16.0 - ~/.nvm/versions/node/v10.15.0/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.15.0/bin/npm
  Browsers:
    Firefox: 67.0.3
    Safari: 12.0
  npmGlobalPackages:
    verdaccio: 4.0.4-fix-info
```
It starts the server and display info at the same time. Solution here is run the Promise synchronously.

Reply:
Here an example that works `./src/cli.js`. It requires a bit of refactoring, this is a minor one, create a `if / else` condition block and only start `init()` if there is no arguments and an specific scenario (who knows why that still there but that's a different story). This make your feature works and scalable to new commands. 

```js
#!/usr/bin/env node

/* eslint no-sync:0 */
/* eslint no-empty:0 */
/**
 * @prettier
 */

import path from 'path';
import semver from 'semver';
import { bgYellow, bgRed } from 'kleur';
import {startVerdaccio, listenDefaultCallback} from './bootstrap';
import findConfigFile from './config-path';
import { parseConfigFile } from './utils';

require('pkginfo')(module);

if (process.getuid && process.getuid() === 0) {
  global.console.warn(bgYellow().red('*** WARNING: Verdaccio doesn\'t need superuser privileges. Don\'t run it under root! ***'));
}

const MIN_NODE_VERSION = '6.9.0';

if (semver.satisfies(process.version, `>=${MIN_NODE_VERSION}`) === false) {
 global.console.error(bgRed(`Verdaccio requires at least Node.js ${MIN_NODE_VERSION} or higher, please upgrade your Node.js distribution`));
 process.exit(1);
}

process.title = 'verdaccio';

const logger = require('./logger');
logger.setup(); // default setup

const envinfo = require('envinfo');
const commander = require('commander');
const pkgVersion = module.exports.version;
const pkgName = module.exports.name;

commander
  .option('-i --info', 'prints debugging information about the local environment')
  .option('-l, --listen <[host:]port>', 'host:port number to listen on (default: localhost:4873)')
  .option('-c, --config <config.yaml>', 'use this configuration file (default: ./config.yaml)')
  .version(pkgVersion)
  .parse(process.argv);

function init() {
  let verdaccioConfiguration;
  let configPathLocation;
  const cliListener = commander.listen;

  try {
    configPathLocation = findConfigFile(commander.config);
    verdaccioConfiguration = parseConfigFile(configPathLocation);
    process.title = verdaccioConfiguration.web && verdaccioConfiguration.web.title || 'verdaccio';

    if (!verdaccioConfiguration.self_path) {
      verdaccioConfiguration.self_path = path.resolve(configPathLocation);
    }
    if (!verdaccioConfiguration.https) {
      verdaccioConfiguration.https = {enable: false};
    }

    logger.logger.warn({file: configPathLocation}, 'config file  - @{file}');

    startVerdaccio(verdaccioConfiguration, cliListener, configPathLocation, pkgVersion, pkgName, listenDefaultCallback);
  } catch (err) {
    logger.logger.fatal({file: configPathLocation, err: err}, 'cannot open config file @{file}: @{!err.message}');
    process.exit(1);
  }
}

if (commander.info) {
  // eslint-disable-next-line no-console
	console.log('\nEnvironment Info:');
  (async () => {
    const data = await envinfo.run({
        System: ['OS', 'CPU'],
        Binaries: ['Node', 'Yarn', 'npm'],
        Virtualization: ['Docker'],
        Browsers: ['Chrome', 'Edge', 'Firefox', 'Safari'],
        npmGlobalPackages: ['verdaccio'],
      });
    // eslint-disable-next-line no-console
    console.log(data);
    process.exit(1);
  })();
} else  if (commander.args.length == 1 && !commander.config) {
  // handling "verdaccio [config]" case if "-c" is missing in command line
  commander.config = commander.args.pop();
  init();
} else if (commander.args.length !== 0) {
  commander.help();
} else {
  init();
}

process.on('uncaughtException', function(err) {
  logger.logger.fatal( {
      err: err,
    },
  'uncaught exception, please report this\n@{err.stack}' );
  process.exit(255);
});

```
Separate different options by

Reply:
Or call `process.exit` or `return`.

Otherwise it might become more difficult in the future.

Reply:
@DanielRuf where exactly? 🤔